cpp/include/raft/random/rng_state.hpp

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2018-2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * This file is deprecated and will be removed in release 22.06.
+ * Please use the cuh version instead.
+ */
+
+#ifndef __RNG_STATE_H
+#define __RNG_STATE_H
+
+#pragma once
+
+#include "detail/rng_state.hpp"
```

Comment:
The implementation details of `rng_state` are so trivial that I think we might be okay just exposing them to the public API. I could really go either way with this, though....